workspaces/ui-v2/src/optic-components/pages/diffs/AddEndpointsPage/AddEndpointsPage.tsx

```diff
@@ -37,128 +26,131 @@ import { useChangelogStyles } from '../../../changelog/ChangelogBackground';
-                  setFilteredUrls(urls.filter((i) => i.path.startsWith(query)));
-                }}
-              />
-              <div style={{ marginLeft: 13 }}>
-                <Typography
-                  variant="subtitle2"
-                  component="span"
-                  color="primary"
-                  style={{ fontWeight: 600, fontSize: 11 }}
-                >
-                  {' '}
-                  bulk mode
-                </Typography>
-                <Switch
-                  value={bulkMode}
-                  onChange={(e: any) => setBulkMode(e.target.checked)}
-                  color="primary"
-                  size="small"
-                />{' '}
-              </div>
-            </Box>
-          </DiffHeader>
-
-          <div style={{ flex: 1 }}>
-            {shownUrls.length > 0 ? (
-              <AutoSizer>
-                {({ height, width }: any) => (
-                  <FixedSizeList
-                    height={height}
-                    width={width}
-                    itemSize={47}
-                    itemCount={shownUrls.length}
-                  >
-                    {renderRow}
-                  </FixedSizeList>
-                )}
-              </AutoSizer>
-            ) : unmatchedUrlLengths === 0 ? (
-              <div className={classes.noResultsContainer}>
-                <Check fontSize="large" />
-                All observed endpoints have been documented
-              </div>
-            ) : (
-              <div className={classes.noResultsContainer}>
-                No urls match the current filter
-              </div>
-            )}
-          </div>
-          <AuthorIgnoreRules />
-        </>
-      }
-      right={<DocumentationRootPageWithPendingEndpoints />}
-    />
+    <>
+      <TwoColumnFullWidth
+        left={
+          <>
+            <AddEndpointDiffHeader
+              searchQuery={searchQuery}
+              setSearchQuery={setSearchQuery}
+              bulkMode={bulkMode}
+              setBulkMode={setBulkMode}
+              numberOfUnmatchedUrl={unmatchedUrlLengths}
+              numberOfVisibleUrls={visibleUrls.length}
+              numberOfSelectedUrls={bulkSelectedUrls.length}
+              checkboxState={checkboxState}
+              toggleSelectAllCheckbox={toggleSelectAllCheckbox}
+              setShowBulkModal={setShowBulkModal}
+            />
+            <div style={{ flex: 1 }}>
+              {visibleUrls.length > 0 ? (
+                <AutoSizer>
+                  {({ height, width }: any) => (
+                    <FixedSizeList
+                      height={height}
+                      width={width}
+                      itemSize={47}
+                      itemCount={visibleUrls.length}
+                      itemKey={(index, data) => {
+                        const item = data.undocumentedUrls[index];
+                        return item.method + item.path;
+                      }}
```

Comment:
So this was the magic piece that we needed to add if the list of urls changes. This allows FixedSizeList to figure out when something is changed, so we don't bleed / reference state between renders + the list changing. 1 more point towards reading documentation